overlord/state/task.go

```diff
@@ -163,13 +167,48 @@ func (t *Task) SetProgress(cur, total int) {
 	t.progress = progress{Current: cur, Total: total}
 }
 
+const (
+	// Messages logged in tasks are guaranteed to use the following strings
+	// plus ": " as a prefix, so these may be handled programatically and
+	// stripped for presentation.
+	LogInfo  = "INFO"
+	LogError = "ERROR"
+)
+
+func (t *Task) logf(kind, format string, args ...interface{}) {
+	if len(t.log) > 9 {
```

Comment:
Yes, but I doubt they'll be simpler, and copying 10 pointers when logging something is nothing compared to running Sprintf. Let's please make this work and learn about the real usage patterns we will develop before investing further.